plugins/processors/lookup/README.md

```diff
@@ -0,0 +1,140 @@
+metric is passed-trough unchanged. By default all matching tags are added and
+existing tag-values are overwritten.
+
+## Global configuration options <!-- @/docs/includes/plugin_config.md -->
+
+In addition to the plugin-specific configuration settings, plugins support
+additional global and plugin configuration settings. These settings are used to
+modify metrics, tags, and field or create aliases and configure ordering, etc.
+See the [CONFIGURATION.md][CONFIGURATION.md] for more details.
+
+[CONFIGURATION.md]: ../../../docs/CONFIGURATION.md#plugins
+
+## Configuration
+
+```toml @sample.conf
+# Lookup a key derived from metrics in a static file
+[[processors.lookup]]
+  ## List of files containing the lookup-table
+  files = ["path/to/lut.json", "path/to/another_lut.json"]
+
+  ## Format of the lookup file(s)
+  ## Available formats are:
+  ##    json               -- JSON file with 'key: {tag-key: tag-value, ...}' mapping
+  ##    csv_key_name_value -- CSV file with 'key,tag-key,tag-value,...,tag-key,tag-value' mapping
+  ##    csv_key_values     -- CSV file with a header and 'key,tag-value,...,tag-value' mapping
+  # format = "json"
+
+  ## Template for generating the lookup-key from the metric.
+  ## This is a Golang template (see https://pkg.go.dev/text/template) to
+  ## access the metric name (`{{.Name}}`), a tag value (`{{.Tag "name"}}`) or
+  ## a field value (`{{.Field "name"}}`).
+  key = '{{.Tag "host"}}'
+```
+
+## File formats
+
+The following descriptions assume `key`s to be unique identifiers used for
+matching the configured `key`. The `tag-name`/`tag-value` pairs are the tags
+added to a metric if the key matches.
+
+### `json` format
+
+In the `json` format, the input `files` must have the following format
+
+```json
+{
+  "keyA": {
+    "tag-name1": "tag-value1",
+    ...
+    "tag-nameN": "tag-valueN",
+  },
+  ...
+  "keyZ": {
+    "tag-name1": "tag-value1",
+    ...
+    "tag-nameM": "tag-valueM",
+  }
+}
+```
+
+Please note that only _strings_ are supported for all elements.
+
+### `csv_key_name_value` format
+
+The `csv_key_name_value` format specifies comma-separated-value files with
+the following format
+
+```csv
+# Optional comments
+keyA,tag-name1,tag-value1,...,tag-nameN,tag-valueN
+keyB,tag-name1,tag-value1
+...
+keyZ,tag-name1,tag-value1,...,tag-nameM,tag-valueM
+```
+
+The formatting uses colons (`,`) as separators and allows for comments defined
+as lines starting with a hash (`#`). All lines can have different numbers but
+must at least contain three columns and follow the name/value pair format, i.e.
+there cannot be a name without value.
+
+### `csv_key_values` format
+
+This setting specifies comma-separated-value files with the following format
+
+```csv
+# Optional comments
+ignored,tag-name1,...,tag-nameN
+keyA,,tag-value1,...,tag-valueN
+keyB,tag-name1,tag-value1,,,,...,
+...
+keyZ,tag-name1,tag-value1,...,tag-nameM,tag-valueM,...,
+```
```

Comment:
Still not sure I follow this example, should tag-name even be in the rows or only the header?